Libs/MRML/Core/vtkMRMLTransformableNode.cxx

```diff
@@ -141,6 +141,45 @@ void vtkMRMLTransformableNode::ProcessMRMLEvents ( vtkObject *caller,
     }
 }
 
+//---------------------------------------------------------------------------
+void vtkMRMLTransformableNode
+::OnNodeReferenceAdded(vtkMRMLNodeReference* reference)
+{
+  Superclass::OnNodeReferenceAdded(reference);
+  if (std::string(reference->GetReferenceRole()) == this->TransformNodeReferenceRole)
+    {
+    this->InvokeCustomModifiedEvent(
+      vtkMRMLTransformableNode::TransformModifiedEvent, reference->GetReferencedNode());
+    }
+
+  vtkMRMLTransformNode* transformNode =
+    vtkMRMLTransformNode::SafeDownCast(reference->GetReferencedNode());
+  if (transformNode)
+    {
+    transformNode->InvokeEvent(
+      vtkMRMLTransformNode::TransformReferenceModifiedEvent, this);
```

Comment:
Have you looked into implementing this in a more generic way, at the vtkMRMLScene level? What I don't like about how it is implemented now that if we want to know if referencing nodes are changed for other node references then we would need to reimplement that from scratch. In contrast, if we implement notification about node reference changes at the scene level, then that mechanism can be used in the future for all kinds of node references.